include/eye.hpp

```diff
@@ -85,191 +80,162 @@ class Eye
+    static constexpr const char* path_image_reflex_5 = "/normal_kxr_reflex.png";
+    static constexpr const char* path_image_upperlid_5 = "/normal_kxr_upperlid_rdown.jpg";
+
+    static constexpr const char* path_image_eyeball_6 = "/normal_kxr_eyeball.jpg";
+    static constexpr const char* path_image_iris_6 = "/white.jpg";
+    static constexpr const char* path_image_pupil_6 = "/white.png";
+    static constexpr const char* path_image_reflex_6 = "/white.png";
+    static constexpr const char* path_image_upperlid_6 = "/normal_kxr_upperlid_happy.jpg";
+
+    using image_path_set = std::array<const char*, 5>;
+    std::map<int, image_path_set> image_path_map;
+    std::map<int, std::vector<float>> upperlid_position_map;
     
-    sprite_upperlid.pushSprite(&sprite_eye, 0, 0, TFT_WHITE);
-
-    sprite_eye.pushRotateZoom(&lcd, lcd.width() >> 1, lcd.height() >> 1, 0, zoom_ratio, zoom_ratio, TFT_WHITE);
-  }
 
-  // 嬉しい目の描画
-  void happy(float dx = 0.0, float dy = 0.0, int happy_level = 0, float scale = 10.0, float random_scale = 5.0)
-  {
-    long rx = (int)(random_scale * random(100) / 100);
-    long ry = (int)(random_scale * random(100) / 100);
-    
-    sprite_eye.clear();
-    sprite_eye.fillScreen(TFT_WHITE);
-    sprite_eyeball.pushSprite(&sprite_eye, 0, 0, TFT_WHITE);
+public:
+    // コンストラクタで EyeManager の参照を受け取る
+    EmotionManager(EyeManager& manager) : eyeManager(manager) {
+      image_path_map = {
+        {0, {path_image_eyeball, path_image_iris, path_image_pupil, path_image_reflex, path_image_upperlid}},
+        {1, {path_image_eyeball_1, path_image_iris_1, path_image_pupil_1, path_image_reflex_1, path_image_upperlid_1}},
+        {2, {path_image_eyeball_2, path_image_iris_2, path_image_pupil_2, path_image_reflex_2, path_image_upperlid_2}},
+        {3, {path_image_eyeball_3, path_image_iris_3, path_image_pupil_3, path_image_reflex_3, path_image_upperlid_3}},
+        {4, {path_image_eyeball_4, path_image_iris_4, path_image_pupil_4, path_image_reflex_4, path_image_upperlid_4}},
+        {5, {path_image_eyeball_5, path_image_iris_5, path_image_pupil_5, path_image_reflex_5, path_image_upperlid_5}},
+        {6, {path_image_eyeball_6, path_image_iris_6, path_image_pupil_6, path_image_reflex_6, path_image_upperlid_6}}
+      };
+
+      upperlid_position_map = {
+        {0, {-130,}},
+        {1, {-130, -130, 0, 0, -130, -130}},
+        {2, {- 130, - 130, -130, -130, 0, 0, -130, -130, -130, -130, 0, -130, -130, 0, -130, -130}},
+        {3, {- 70, - 70, -60, 0, 0, 0, -60, -70, -70, - 70}},
+        {4, {- 10}},
+        {5, {0}},
+        {6, {0}},
+      };  
+    }
     
-    sprite_upperlid.pushSprite(&sprite_eye, rx, ry, TFT_WHITE);
+    int frame_index = 0;
 
-    sprite_eye.pushRotateZoom(&lcd, lcd.width() >> 1, lcd.height() >> 1, 0, zoom_ratio, zoom_ratio, TFT_WHITE);
-  }
-};
+    // 感情ごとの表示設定
+    void set_emotion(int eye_status, bool& emotion_changed_flag, int frame = 0){
```

Comment:
We don't have to care if emotion is changed or not in out of this class if we have `EmotionManger.eye_status` member variable and check if input `eye_status` changed from it.

Reply:
I feel like this methot does multiple process at the same time (1. set emotion status, 2. update eye configuration).
So we can split them into several process (`set_emotion(int eye_status)` and `update(int frame)`)

Reply:
And if we have `frame` count internally, we don't have to care it out of class.